### ISSUE TITLE
PSR12.Files.ImportStatement should not apply to traits

### ISSUE THREAD

**Issue by chrisminett** [squizlabs/PHP_CodeSniffer#2665]
The `PSR12.Files.ImportStatement` sniff is being applied to `use` statements within a class, where they are used to implement traits. The sniff should not be applied in this scenario.

The PSR-12 documentation in [section 3](https://www.php-fig.org/psr/psr-12/#3-declare-statements-namespace-and-import-statements) applies to import statements before the class definition as described in the list at the top of the section (class-based, function-based, constant-based).

Traits are documented separately in [section 4.2](https://www.php-fig.org/psr/psr-12/#42-using-traits) and describe the requirements for the `use` keyword to implement traits, but does not require that the value does not start with a backslash.

h2. Current behaviour

```php
<?php

namespace MyNamespace;

class MyClass
{
    use \MyTrait;
}
```

```
 $ phpcs --standard=PSR12 test.php                                                                     -1-

FILE: test.php
---------------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
---------------------------------------------------------------------------------
 7 | ERROR | Import statements must not begin with a leading backslash
---------------------------------------------------------------------------------

```

h2. Expected behaviour

With the same example, no error is found.

**Comment by gsherwood** [comment]
Thanks for reporting this. Fix will be in 3.5.2.
